pkg/controller/installation/products/codeready/reconciler_test.go

```diff
@@ -62,6 +83,380 @@ func TestCodeready(t *testing.T) {
 				},
 			},
 		},
+		{
+			Name:                 "test error on bad RHSSO config",
+			ExpectedStatus:       v1alpha1.PhaseNone,
+			ExpectedCreateError:  "keycloak config is not valid: config realm is not defined",
+			Object:               &v1alpha1.Installation{},
+			FakeControllerClient: pkgclient.NewFakeClient(),
+			FakeConfig: &config.ConfigReadWriterMock{
+				ReadCodeReadyFunc: func() (ready *config.CodeReady, e error) {
+					return config.NewCodeReady(config.ProductConfig{}), nil
+				},
+				ReadRHSSOFunc: func() (*config.RHSSO, error) {
+					return config.NewRHSSO(config.ProductConfig{}), nil
+				},
+			},
+		},
+		{
+			Name:                 "test awaiting subscrit on failed RHSSO config",
```

Comment:
subscrit ?

Reply:
"subscrit on" = "subscription" lol, I'll update it!